src/components/PrimaryNav.tsx

```diff
@@ -1,23 +1,137 @@
-import React from 'react';
-import { Menu } from '@headlessui/react';
-import logo from '../assets/logo.svg';
-import useSupabaseAuth from '../hooks/useSupabaseAuth';
-import { version } from '../../package.json';
-import { capturePostHogAnayltics } from '../lib/analytics';
+import React, {useState, useEffect, ChangeEvent} from "react";
+import { Menu } from "@headlessui/react";
+import logo from "../assets/logo.svg";
+import useSupabaseAuth from "../hooks/useSupabaseAuth";
+import { version } from "../../package.json";
+import { fetchWithSearch } from "../lib/supabase";
 
-const PrimaryNav = (): JSX.Element => {
+interface PostWrapProps{
+  setTextToSearch: any
+}
+
+// TODO: move to hooks/debounce.ts
```

Comment:
use https://usehooks.com/useDebounce/ please or the radix example one :D

Reply:
Thanks for reviewing, This hook will be called only when the user clicks on the repository item from the input dropdown items. So, it is not going to be called frequently. Do we still need the denouncing here?